File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_set = {}
        for i in range(len(nums)):
            new_target = target - nums[i]
            if new_target in hash_set:
                return [hash_set.get(new_target),i]
            hash_set[nums[i]] = i

File: two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        vector<int> result;
        for(int i=0;i<nums.size();++i){
            int complimentary = target - nums[i];
            if (hash.find(complimentary) != hash.end()){
                result.push_back(hash[complimentary]);
                result.push_back(i);
                return result;
            }
            hash[nums[i]] = i;                
        }
        return nums;
    }
};